utils.go

```diff
@@ -15,23 +44,139 @@ type Storage struct {
 
 // String implements Storager.String
 func (s *Storage) String() string {
-	panic("implement me")
+	return fmt.Sprintf("Storager us3 {Name: %s, WorkDir: %s", s.bucket, s.workDir)
+}
+
+func New(pairs ...types.Pair) (types.Servicer, types.Storager, error) {
+	return newServicerAndStorager(pairs...)
+}
+
+func NewServicer(pairs ...types.Pair) (types.Servicer, error) {
+	return newServicer(pairs...)
 }
 
 // NewStorager will create Storager only.
 func NewStorager(pairs ...types.Pair) (types.Storager, error) {
-	return newStorager(pairs...)
+	_, store, err := newServicerAndStorager(pairs...)
+	return store, err
 }
 
-// newStorager will create a storage client.
-func newStorager(pairs ...types.Pair) (store *Storage, err error) {
+func newServicer(pairs ...types.Pair) (srv *Service, err error) {
 	defer func() {
 		if err != nil {
-			err = services.InitError{Op: "new_storager", Type: Type, Err: formatError(err), Pairs: pairs}
+			err = services.InitError{
+				Op:    "new_servicer",
+				Type:  Type,
+				Err:   formatError(err),
+				Pairs: pairs,
+			}
 		}
 	}()
 
-	panic("implement me")
+	srv = &Service{}
+
+	opt, err := parsePairServiceNew(pairs)
+	if err != nil {
+		return nil, err
+	}
+
+	cp, err := credential.Parse(opt.Credential)
+	if err != nil {
+		return nil, err
+	}
+	if cp.Protocol() != credential.ProtocolHmac {
+		return nil, services.PairUnsupportedError{Pair: ps.WithCredential(opt.Credential)}
+	}
+	ak, sk := cp.Hmac()
+
+	ep, err := endpoint.Parse(opt.Endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	var url string
+	switch ep.Protocol() {
+	case endpoint.ProtocolHTTP:
+		url, _, _ = ep.HTTP()
+	case endpoint.ProtocolHTTPS:
+		url, _, _ = ep.HTTPS()
+	default:
+		return nil, services.PairUnsupportedError{Pair: ps.WithEndpoint(opt.Endpoint)}
+	}
+
+	config := &us3.Config{
+		PublicKey:  ak,
+		PrivateKey: sk,
+		FileHost:   url,
+		BucketHost: "api.ucloud.cn",
+	}
+
+	srv.service, err = us3.NewFileRequest(config, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if opt.HasDefaultServicePairs {
+		srv.defaultPairs = opt.DefaultServicePairs
+	}
+	if opt.HasServiceFeatures {
+		srv.features = opt.ServiceFeatures
+	}
+
+	return
+}
+
+func newServicerAndStorager(pairs ...types.Pair) (srv *Service, store *Storage, err error) {
+	srv, err = newServicer(pairs...)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	store, err = srv.newStorage(pairs...)
+	if err != nil {
+		err = services.InitError{Op: "new_storager", Type: Type, Err: formatError(err), Pairs: pairs}
+		return nil, nil, err
+	}
+
+	return
+}
+
+func (s *Service) newStorage(pairs ...types.Pair) (store *Storage, err error) {
+	opt, err := parsePairStorageNew(pairs)
+	if err != nil {
+		return nil, err
+	}
+
+	store = &Storage{
+		client:  s.service,
```

Comment:
`BucketName` should be specified for `NewFileRequest`, ref: https://github.com/ufilesdk-dev/ufile-gosdk/blob/master/request.go#L48-L49.

Reply:
```
//Request 创建后的 instance 不是线程安全的，如果你需要做并发的操作，请创建多个 UFileRequest。
```

It sound like we need to store a `http.Client` here and create `UFileRequest` every time we sending API.